ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -161,14 +163,21 @@ bracketChainSyncClient ::
  -> StrictTVar m (Map peer (StrictTVar m (AnchoredFragment (Header blk))))
     -- ^ The candidate chains, we need the whole map because we
     -- (de)register nodes (@peer@).
+ -> StrictTVar m (Set peer)
+    -- ^ The upstream peers that have not claimed any headers subsequent to
+    -- their candidate, aka those that are /idle/.
```

Comment:
My intuition is to avoid "unnecessary" `TVar`s.

-----

Are you sure we'll need to add more? I'm not recalling why.

> Given that we might want to add even more information here in the future (at least peers that are stuck due to the forecasting horizon?), should this be bundled in some type?

Reply:
> > Given that we might want to add even more information here in the future (at least peers that are stuck due to the forecasting horizon?), should this be bundled in some type?
> 
> Are you sure we'll need to add more? I'm not recalling why.

I think I thought about the LoP (as we have to pause it while being waiting for the forecasting horizon to advance), but I don't think we have to record any information in `TVar`s for that :+1: